stix_shifter/utils/error_response.py

```diff
@@ -0,0 +1,124 @@
+        # document it: '+' and '~'
+        if message_struct is not None and message_path is not None:
+            if (isinstance(message_struct, collections.Mapping) or type(message_struct).__name__=='list'):
+                struct = message_struct.copy()
+                for i in message_path:
+                    if (isinstance(struct, collections.Mapping) and i in struct) or (type(struct).__name__=='list' and isinstance(i, int) and i < len(struct)):
+                        struct = struct[i]
+                        if struct is None:
+                            break
+                    elif i[:1] == '+' and isinstance(struct, list):
+                        key, value = i[1:].split('=')
+                        filted_struct = list(filter(lambda item: (key in item and str(item[key])==str(value)), struct))
+                        struct = filted_struct
+                    elif i[:1] == '~' and isinstance(struct, list):
+                        key = i[1:]
+                        filtered_struct = set()
+                        for s in struct:
+                            filtered_struct.add(s[key])
+                        struct = list(filtered_struct)
+                    else:
+                        break
+                return struct
+            else:
+                return message_struct
+
+    @staticmethod
+    def fill_error(return_object, message_struct=None, message_path=None, message=None, error=None):
+        return_object['success'] = False
+        error_code = ErrorCode.TRANSMISSION_UNKNOWN
+        
+        if message is None:
+            message = ''
+
+        struct_item = ErrorResponder.get_struct_item(message_struct, message_path)
+        if struct_item is not None:
+            if len(message) > 0:
+                message += ';'
+            if (isinstance(struct_item, list)):
+                struct_item = json.dumps(struct_item)
+            message += str(struct_item)
+        error_msg = ''
+        if error is not None:
+            str_error = str(error)
+            #TODO replace with logger + stacktrace it to logger
+            print("error occurred: " + str_error)
+            if isinstance(error, SSLError):
+                error_code = ErrorCode.TRANSMISSION_AUTH_SSL
+                error_msg = 'Wrong certificate'
+            elif isinstance(error, ConnectionError):
+                error_code = ErrorCode.TRANSMISSION_CONNECT
+                error_msg = 'Connection error'
+            else:
+                error_msg = str(error)
+
+            if len(error_msg) > 0:
+                if len(message) > 0:
+                    message += '; '
+                message += error_msg
+
+        if message is not None and len(message)>0:
+            if error_code.value == ErrorCode.TRANSMISSION_UNKNOWN.value:
+                if 'uthenticat' in message or 'uthoriz' in message:
```

Comment:
authenticate* authorize* ?

Reply:
upper case and lower case for the first letter with one pattern